flow-server/src/main/java/com/vaadin/flow/server/communication/WebComponentBootstrapHandler.java

```diff
@@ -573,8 +573,8 @@ private static String inlineHTML(String html) {
                 .replace("//<![CDATA[", "/*<![CDATA[*/")
                 .replace("//]]>", "/*]]>*/")
                 // get rid of all the unnecessary white-space
-                .replaceAll("\\s{2,}", "").replace("\t", "").replace("\n", "")
-                .replace("\r", "");
+                .replaceAll("\\s{2,}", "").replace("\t", "")
+                .replace("\n", "\\n").replace("\r", "");
```

Comment:
```suggestion
                .replace("\r", "").replaceAll("\\s{2,}", "")
                .replace("\t", "").replace("\n", "\\n");
```
Needs to be changed so that any \r is removed first as on windows its \r\n for new line and that falls into the \s for 2 characters or more